crypto/sleep.c

```diff
@@ -28,13 +28,12 @@ void OSSL_sleep(uint64_t millis)
 
     /* HPNS does not support usleep for non threaded apps */
     PROCESS_DELAY_(millis * 1000);
-#  elif defined(_SPT_MODEL_)
-#   include <spthread.h>
-#   include <spt_extensions.h>
-
-    usleep(millis * 1000);
 #  else
-    usleep(millis * 1000);
+    unsigned int s = (unsigned int)(millis / 1000);
+    unsigned int us = (unsigned int)((millis % 1000) * 1000);
+
+    sleep(s);
+    usleep(us);
```

Comment:
This seems identical to the `#else` code below. I wonder if it could be collapsed together? e.g. by making line 25 above

```` C
# elif defined(__TANDEM) && !defined(_REENTRANT)
````

Reply:
I'm happy to do that. The last time I submitted this for the original fix, it was recommended to split the #ifdef. Will revise.

Reply:
Thank you @rsbeckerca I will approve as soon as you push the requested changes.  Please note that I've marked this as a regression, and our beta freeze is impending on [tuesday, march 26](https://github.com/orgs/openssl/projects/11/views/22).  Given that this will need a 24 hour clock between approval and merge, I would greatly appreciate it if you could get this posted today so we can start that clock running, and make the deadline without the need for additional process gymnastics.

Reply:
I have just pushed the requested fix. This fix is applicable to master and the entire 3.x set of releases.